ArgusCore/src/main/java/com/salesforce/dva/argus/entity/InfractionHistory.java

```diff
@@ -0,0 +1,217 @@
+package com.salesforce.dva.argus.entity;
+
+import java.util.List;
+import javax.persistence.Basic;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.EntityManager;
+import javax.persistence.FetchType;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.NamedQueries;
+import javax.persistence.NamedQuery;
+import javax.persistence.NoResultException;
+import javax.persistence.Table;
+import javax.persistence.TypedQuery;
+
+/**
+ * The entity encapsulates information about the infraction history.
+ *
+ * <p>Fields that cannot be null are:</p>
+ *
+ * <ul>
+ *   <li>POLICY_ID</li>
+ *   <li>USER_ID</li>
+ *   <li>INFRACTION_TIMESTAMP</li>
+ *   <li>EXPIRATION_TIMESTAMP</li>
+ * </ul>
+ *
+ * @author  Ruofan Zhang (rzhang@salesforce.com)
+ */
+@SuppressWarnings("serial")
+@Entity
+@Table(name = "INFRACTION_HISTORY")
+@NamedQueries(
+	    {
+	        @NamedQuery(
+	            name = "InfractionHistory.findByPolicyIdAndUserName", 
+	            query = "SELECT r FROM InfractionHistory r WHERE r.policy = :policy AND r.user in (SELECT p.id from PrincipalUser p where p.userName = :userName)"
```

Comment:
Select r from infraction r, user u where r.policy = :policy AND r.user = u.id and u.userName = :userName;